src/tike/ptycho/probe.py

```diff
@@ -285,10 +283,12 @@ def simulate_varying_weights(scan, eigen_probe):
 
 def init_varying_probe(scan, shared_probe, N):
     """Initialize arrays for N eigen modes."""
+    if N < 1:
+        return None, None
 
     eigen_probe = tike.random.numpy_complex(
         *shared_probe.shape[:-4],
-        N,
```

Comment:
Why change to N-1?

Reply:
The 0th probe and the 0th variable/varying probe are now the same, so there are only N-1 indices in the the varying probe array. i.e. If you want 1 variable probe, then we don't need a separate array for it because it is already saved in the probe array.